gnmi/gnmit/collector.go

```diff
@@ -0,0 +1,114 @@
+package gnmit
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/openconfig/gnmi/cache"
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/openconfig/gnmi/subscribe"
+)
+
+var (
+	// metadataUpdatePeriod is the period of time after which the metadata for the collector
+	// is updated to the client.
+	metadataUpdatePeriod = 30 * time.Second
+	// sizeUpdatePeriod is the period of time after which the storage size information for
+	// the collector is updated to the client.
+	sizeUpdatePeriod = 30 * time.Second
+)
+
+// Collector is a basic gNMI target that supports only the Subscribe
+// RPC, and acts as a cache for exactly one target.
+type Collector struct {
+	cache *cache.Cache
+	// name is the hostname of the client.
+	name string
+	// inCh is a channel use to write new SubscribeResponses to the client.
+	inCh chan *gpb.SubscribeResponse
+	// stopFn is the function used to stop the server.
+	stopFn func()
+}
+
+// NewCollector returns an initialized Collector.
+func NewCollector(hostname string) *Collector {
```

Comment:
maybe add line saying, something like "To create server that supports gnmi.Set, use New() instead"